### ISSUE TITLE
Change from `optional` to `require-optional`

### ISSUE THREAD

**Issue by baflo** [nestjs/nest#3035]
## Is your feature request related to a problem? Please describe.
The package [`optional`](https://www.npmjs.com/package/optional) requires packages relative to its own location, which may result in errors, if the application has a structure like this:

```
+ node_modules
| + optional
| + my-nest-server
|    + node_modules
|    | + @nestjs/common
|    | + @nestjs/core
|    | + @nestjs/websockets
|    + package.json
|    + server.js
+ app.js
+ package.json
```

Here, `@nestjs/websockets` won't be loaded, as it cannot be found by `optional`. Hence, nestjs won't initialize web-socket. 

Some issues are also filed: #2213, #1149

## Describe the solution you'd like

The package [`require-optional`](https://www.npmjs.com/package/require-optional) can be used as a dropin-replacement and resolves this issue, as it requires relative to the `__filename` the require was called.

## Teachability, Documentation, Adoption, Migration Strategy
This can be used as a dropin replacement and shouldn't required special training.

## What is the motivation / use case for changing the behavior?
Resolving filed issues and others...

**Comment by kamilmysliwiec** [comment]
This PR will solve this issue https://github.com/nestjs/nest/pull/3041

**Comment by kamilmysliwiec** [comment]
Published as 6.8.0
